project-base/storefront/helpers/products/LastVisitedProducts.tsx

```diff
@@ -0,0 +1,34 @@
+import { usePersistStore } from 'store/zustand/usePersistStore';
+
+const LAST_VISITED_MAX_ITEMS = 20;
+
+export const GetLastVisitedProductCatalogNumbers = (): string[] => {
+    const lastVisitedProducts = usePersistStore((store) => store.lastVisitedProducts);
+    return lastVisitedProducts ? JSON.parse(lastVisitedProducts) : [];
+};
+
+export const SetLastVisitedProductCatalogNumbers = (newLastVisitedItemCatalogNumber: string): void => {
+    let olderItemCatalogNumbers = GetLastVisitedProductCatalogNumbers();
+
+    const updateUserState = usePersistStore((store) => store.updateUserState);
+    const alreadyAdded = olderItemCatalogNumbers.some(
+        (oldItemCatalogNumber) => oldItemCatalogNumber === newLastVisitedItemCatalogNumber,
+    );
```

Comment:
If it's just an array of strings, you can use `includes`, as it is more semantically clear and the performance should also be a tiny bit better, though this is negligible for sure in this case.